ccpp/data/GFS_typedefs.F90

```diff
@@ -1161,27 +1161,25 @@ module GFS_typedefs
     logical              :: do_shum
     logical              :: do_skeb
     integer              :: skeb_npass
-    integer              :: lndp_type
+    integer              :: lndp_type         ! integer indicating land perturbation scheme type:
+                                              ! 0 - none
+                                              ! 1 - scheme from Gehne et al, MWR, 2019.  (Noah only, not maintained?)
+                                              ! 2 - scheme from Draper, JHM, 2021.
     real(kind=kind_phys) :: sppt_amp          ! pjp cloud perturbations
     integer              :: n_var_lndp
-    logical              :: lndp_each_step    ! flag to indicate that land perturbations are applied at every time step,
-                                              ! otherwise they are applied only after gcycle is run
-    character(len=3)    , pointer :: lndp_var_list(:)  ! dimension here must match  n_var_max_lndp in  stochy_nml_def
-    real(kind=kind_phys), pointer :: lndp_prt_list(:)  ! dimension here must match  n_var_max_lndp in  stochy_nml_def
-                                              ! also previous code had dimension 5 for each pert, to allow
-                                              ! multiple patterns. It wasn't fully coded (and wouldn't have worked
-                                              ! with nlndp>1, so I just dropped it). If we want to code it properly,
-                                              ! we'd need to make this dim(6,5).
+    ! next two are duplicated here to support lndp_type=1. If delete that scheme, could remove from GFS defs?
+    character(len=3)    , pointer :: lndp_var_list(:)
+    real(kind=kind_phys), pointer :: lndp_prt_list(:)
     logical              :: do_spp            ! Overall flag to turn on SPP or not
     integer              :: spp_pbl
     integer              :: spp_sfc
     integer              :: spp_mp
     integer              :: spp_rad
     integer              :: spp_gwd
     integer              :: n_var_spp
-    character(len=3)    , pointer :: spp_var_list(:)  ! dimension here must match n_var_spp in stochy_nml_def
-    real(kind=kind_phys), pointer :: spp_prt_list(:)  ! dimension here must match n_var_spp in stochy_nml_def 
-    real(kind=kind_phys), pointer :: spp_stddev_cutoff(:)  ! dimension here must match n_var_spp in stochy_nml_def 
+    character(len=3)    , pointer :: spp_var_list(:) 
```

Comment:
Is it still true that the spp_var_list needs to have dimension n_var_spp?

Reply:
Yes, but I don't think the comment is required anymore. It looks to me like these lines have been copied from my old lndp_var_list and lndp_prt_list code and then adapted to spp. Originally lndp_var_list and lndp_prt_list were specified here as arrays and not pointers. The same array is also specified in stochastic_physics, so the dimension of the array here had to match that specified in stochastic physics - hence I added the comment. Now that lndp_var/prt_list and spp_var/prt_list are pointers I don't think we need the comment anymore.